### DISCUSSION TITLE
SSE in routes and `Task timed out after 10.01 seconds`

### DISCUSSION THREAD

**Discussion by blooser** [vercel/community#7238] [Help] [answered]
### Summary

Hello!

I've got a problem while using eventsource in my api endpoint. Next.js app is connecting with another app using eventsource and keep connection alive but on vercel cloud it is killed and exception is throwed `Task timed out after 10.01 seconds`.

I suppose it happens because of restrictions on serverless function invoking - but is there any possible way to keep SSE working on routes on the vercel cloud?

I'm using https://www.npmjs.com/package/eventsource for SSE on the server side.

### Steps to Reproduce

To reproduce just connect to this endpoint on the frontend side using EventSource while being deployed on the Vercel cloud.

```js
import { routes } from '@/src/routes';
import EventSource from 'eventsource';
import type { NextApiResponse } from 'next';
import { cookies } from 'next/headers';
import { NextRequest } from 'next/server';
import { NextResponse } from 'next/server';

export async function GET(req: NextRequest, res: NextApiResponse)
    console.log('Listening for events');

    let responseStream = new TransformStream();
    const writer = responseStream.writable.getWriter();
    const encoder = new TextEncoder();

    const eventSourceUrl = `${routes.backend.events}`;
    const eventSource = new EventSource(eventSourceUrl, {
        headers: { Cookie: cookies().toString() },
    });

    eventSource.onopen = (e) => {
        console.log('Connection established');
    };

    eventSource.onmessage = async (e) => {
        try {
            console.log(`Received message: ${e.data}`);
            await writer.write(encoder.encode(e.data));
        } catch (err) {
            console.error('Error writing to stream', err);
        }
    };

    eventSource.onerror = async (e) => {
        console.error('ERROR', e);

        await writer.ready;
        await writer.close();

        eventSource.close();
    };

    req.signal.onabort = async () => {
        console.log('Client disconnected. Closing EventSource');
        await writer.ready;
        await writer.close();

        eventSource.close();
    };

    return new NextResponse(responseStream.readable, {
        status: 200,
        headers: {
            Connection: 'keep-alive',
            'Cache-Control': 'no-cache, no-transform',
            'Content-Encoding': 'none',
            'Content-Type': 'text/event-stream',
        },
    });
}
```

**Comment by leerob** [comment]
Hey there, have you seen this? https://vercel.com/docs/functions/configuring-functions/duration

**Reply by blooser** [reply]
Changed max duration to 60 and error still occurs `Task timed out after 60.01 seconds`

**Reply by leerob** [reply]
And are you wanting it to progressively stream back responses? https://vercel.com/docs/functions/streaming/streaming-examples

**Reply by blooser** [reply, marked as answer]
Thanks for trying to help, but I found the answer for my problem, the code will not work due to: https://github.com/vercel/next.js/issues/9965#issuecomment-614823642

[a comment by an automated account is not shown]
